vault/resource_database_secret_backend_connection.go

```diff
@@ -752,6 +754,20 @@ func connectionStringResource(config *connectionStringConfig) *schema.Resource {
 		}
 	}
 
+	if config.isCloud {
+		res.Schema["auth_type"] = &schema.Schema{
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Specify alternative authorization type. (Only 'gcp_iam' is valid currently)",
+		}
+		res.Schema["service_account_json"] = &schema.Schema{
```

Comment:
Should we add a Vault Server version check for this and `auth_type`? You can see an examples of doing version checks in this PR: https://github.com/hashicorp/terraform-provider-vault/pull/1936/files#diff-3c100f49c020a28974d882dd5b34d276a7f825e3ccfaa18e584c1e26e2261d10R237. (Looks like you already have checks in the test files ✅ .)

Reply:
Oh yeah, we should, ty

Reply:
I have added the meta interface to a bunch of the data setting calls so we can do version checking - the only one that actually uses it are the two params we added here, but to get that far in, a bunch of calling functions needed to grow a new parameter.

If there's a cleaner approach i missed, i would be pleased to use it.

Reply:
version checking LGTM!